test/src/specs/relay/compact_block.rs

```diff
@@ -152,13 +143,71 @@ impl CompactBlockBasic {
 
         clear_messages(net);
     }
+
+    pub fn test_lose_get_block_transactions(
+        &self,
+        net: &Net,
+        node0: &Node,
+        node1: &Node,
+        peer_id0: PeerIndex,
+    ) {
+        node0.generate_block();
+        let new_tx = node0.new_transaction(node0.get_tip_block().transactions()[0].hash().clone());
+        node0.submit_block(
+            &node0
+                .new_block_builder(None, None, None)
+                .proposal(new_tx.proposal_short_id())
+                .build(),
+        );
+        // Proposal a tx, and grow up into proposal window
+        (0..6).for_each(|_| {
+            node0.generate_block();
+        });
+
+        // Make node0 and node1 reach the same height
+        node1.generate_block();
+        node0.connect(node1);
+        assert!(wait_until(20, || node0.get_tip_block() == node1.get_tip_block()),);
```

Comment:
we can use the helper fn `waiting_for_sync`: 
```suggestion
    node0.waiting_for_sync(node1, 9, 20);
```